mysite/scripts/snapshot_then_push.sh

```diff
@@ -30,16 +30,37 @@ cd "$PREFIX"
 
 ## Create a path to store the JSON in
 TEMPFILE="$(mktemp --suffix -snapshot.json.gz)"
+TEMPDB = "$(mktemp --suffix -snapshot.db)"
 
 ## Do the slooow thing of snapshotting the database
 ./manage.py snapshot_public_data | gzip > "$TEMPFILE"
 
+## Create a new settings to specify the name of the temporary snapshot db
+touch mysite/snapshot_settings.py
+echo "from mysite.settings import *; DATABASES = {
+    'default': {
+        'NAME': os.path.join(MEDIA_ROOT_BEFORE_STATIC, "$TEMPDB"),
+        'ENGINE': 'django.db.backends.sqlite3',
+        'CHARSET': 'utf8',
+    },
+}" > mysite/snapshot_settings.py
+
+# Load in the snapshot data, which automagically generates the temporary db in root directory
+echo "TEMPFILE VARIABLE: " $TEMPFILE
+./manage.py syncdb --noinput --migrate --settings=mysite.snapshot_settings
+./manage.py migrate --settings=mysite.snapshot_settings
+./manage.py loaddata $TEMPFILE --settings=mysite.snapshot_settings
```

Comment:
You should use `""` around your use of bash variables.

See e.g. http://tldp.org/LDP/abs/html/quotingvar.html